vnext/Microsoft.ReactNative/Fabric/Composition/FocusManager.cpp

```diff
@@ -124,8 +124,17 @@ winrt::Microsoft::ReactNative::implementation::ComponentView *NavigateFocusHelpe
       [reason, &toFocus](::winrt::Microsoft::ReactNative::implementation::ComponentView &v) noexcept
       -> bool { return (toFocus = NavigateFocusHelper(v, reason)); };
 
-  if (view.runOnChildren(reason == winrt::Microsoft::ReactNative::FocusNavigationReason::First, fn)) {
-    return toFocus;
+  // Do not iterate through sub rootviews when looking for somewhere to put focus.
+  // If the root view is focusable, we can set focus directly on that, and it can put focus in the  appropriate place.
```

Comment:
Is this going to work as intended for subRootViews that aren't the first child of their parent, or is that not a possible situation?

Reply:
Not entirely sure if this is more for avoiding focus accidentally moving out of the Modal or dealing with focus returning to the app when switching apps?  Also curious how it plays into brownfield apps if `Modal` actually takes modality, or if Modal is realistically a no-go for brownfield usage.

Is the idea this NavigateFocusHelper will help us move focus into a RootView ReactNativeIsland when it's activated, and if it happens to have a `Modal` subRootView then we end up no-op'ing the first/last focusable information on the RootView, and the activation will get processed again by the subRootView?

It seems short of what we'd need for non-modal experiences but makes sense for iteration?

Reply:
Cleaned up the mess by spitting the componentView trees.